comp/core/autodiscovery/integration/config.go

```diff
@@ -8,15 +8,19 @@ package integration
 
 import (
 	"fmt"
+	"io"
 	"sort"
 	"strconv"
 	"strings"
 
+	"github.com/fatih/color"
 	"github.com/twmb/murmur3"
 	yaml "gopkg.in/yaml.v2"
 
+	checkid "github.com/DataDog/datadog-agent/pkg/collector/check/id"
```

Comment:
I think this is wrong thing to import here. AD is a provider data to multiple components, it shouldn't depend on packages from one of its clients.

Would it make sense to move scurbbed report generation to a separate package? Or, alternatively, move check id code into AD package?

Reply:
Perhaps we could avoid creating a circular import if all formatting code lived in `autodiscoveryimpl`?

Reply:
Is not a bad idea, but the problem with the approach is that we would have to include the `autodiscoveryimpl` in the `pkg/flare/cluster_checks.go` 

From the point of view of organization makes sense that the config has the function `PrintConfig` as it is using all the struct fields. I guess we could have a helper package in the autodiscovery component package, that way we can use the PrintConfig outside of the autodiscovery component without importing all the internals of the autodiscovery implementation

Reply:
> From the point of view of organization makes sense that the config has the function PrintConfig as it is using all the struct fields.

You can also consider PrintConfig to be a part of a larger template that needs to produce consistent output, so it could be grouped with the rest of presentation code. Grouping based on purpose, rather than data it accesses. Not that it matters I guess.

Did it help avoid the import cycle, so that https://github.com/DataDog/datadog-agent/pull/25595/commits/b428c446e97406a2e737da8d09b6f7877ed12038 is no longer needed?

Reply:
> You can also consider PrintConfig to be a part of a larger template that needs to produce consistent output, so it could be grouped with the rest of presentation code

The problem with that is that is that the scrubbing logic needs to happen in the server as the scrubber state is updated by the secret resolver component every time a encrypted key is decrypted. 

Prior to my change the presentation code was in the client, but that lead to some secrets not being scrubber. That issue was the main reason to move the presentation code closer to the server.

Reply:
> Did it help avoid the import cycle, so that https://github.com/DataDog/datadog-agent/commit/b428c446e97406a2e737da8d09b6f7877ed12038 is no longer needed?

Yeah, the import cycle is fixed. I can revert that commit

Reply:
Done ✅